zulipterminal/ui_tools/boxes.py

```diff
@@ -393,8 +401,11 @@ def keypress(self, size: urwid_Size, key: str) -> Optional[str]:
             if self.focus_position == self.FOCUS_CONTAINER_HEADER:
                 if self.to_write_box is None:
                     if header.focus_col == self.FOCUS_HEADER_BOX_STREAM:
-                        stream_name = (header[self.FOCUS_HEADER_BOX_STREAM]
-                                       .edit_text)
+                        if not self.in_edit_message_box:
```

Comment:
@prah23 see this portion.

Reply:
@Abhirup-99 I was referring to the `is_in_edit` boolean.

Reply:
We could know that a user is in edit mode on keypress (i.e., `stream_box_edit_view` called). That said, the extra boolean operator could be an unwanted variable (unless you aim to use it for a different purpose) :)

Reply:
@Ezio-Sarthak The extra boolean value is for autocompletion parts, which is completely detached from this place. That's why it was necessary to have that.

Reply:
I agree on a declaration of boolean :+1:. Another approach could be that the user never cycles to the stream box, if in edit mode. Through this, we could skip this conditional check setting `stream_name`. One may think then, that this would throw an error when user tries to send the message, but our `self.stream_write_box` never changes in the whole process (cycling + editing + sending), so it would not be erroneous :)

Reply:
> I agree on a declaration of boolean . Another approach could be that the user never cycles to the stream box, if in edit mode. Through this, we could skip this conditional check setting `stream_name`. One may think then, that this would throw an error when user tries to send the message, but our `self.stream_write_box` never changes in the whole process (cycling + editing + sending), so it would not be erroneous :)
Since the cycling logic is completely detached, it might create more constraints in future implementations. Already the entire logic is a serious of if else, pushing one more on that list doesn't seem to me a good idea.